src/swap/changenow.js

```diff
@@ -131,260 +119,176 @@ export function makeChangeNowPlugin(
         getAddress(request.toWallet, request.toCurrencyCode)
       ])
 
-      // transcribe currencyCodes if necessary
-      const { safeFromCurrencyCode, safeToCurrencyCode } = safeCurrencyCodes(
-        CURRENCY_CODE_TRANSCRIPTION,
-        request
-      )
-
-      // get the markets
-      const availablePairs = await get(`currencies-to/${safeFromCurrencyCode}`)
-      const fixedMarket = asFixedMarketReply(
-        await get(
-          `market-info/fixed-rate/${apiKey}` +
-            (promoCode != null ? `?promo=${promoCode}` : '')
-        )
-      )
-
-      const quoteAmount =
-        request.quoteFor === 'from'
-          ? await request.fromWallet.nativeToDenomination(
-              request.nativeAmount,
-              request.fromCurrencyCode
-            )
-          : await request.toWallet.nativeToDenomination(
-              request.nativeAmount,
-              request.toCurrencyCode
+      const {
+        fromCurrencyCode,
+        toCurrencyCode,
+        fromMainnetCode,
+        toMainnetCode
+      } = getCodes(request, false)
+      const currencyString = `&fromCurrency=${fromCurrencyCode}&toCurrency=${toCurrencyCode}&fromNetwork=${fromMainnetCode}&toNetwork=${toMainnetCode}`
+
+      const fromAmountEntered = request.quoteFor === 'from'
+      const { nativeAmount } = request
+      const largeDenomAmount = fromAmountEntered
+        ? await request.fromWallet.nativeToDenomination(
+            nativeAmount,
+            fromCurrencyCode
+          )
+        : await request.toWallet.nativeToDenomination(
+            nativeAmount,
+            toCurrencyCode
+          )
+
+      // Try fixed-rate first
+      const orderFlows = ['fixed-rate', 'standard']
+
+      for (const flow of orderFlows) {
+        const isFixedRate = flow === 'fixed-rate'
+
+        try {
+          if (!isFixedRate && !fromAmountEntered) {
```

Comment:
split into is buying and isselling functions and each one takes a bunch of params